pkg/readiness/ready_tracker.go

```diff
@@ -333,6 +344,28 @@ func (t *Tracker) collectInvalidExpectations(ctx context.Context) {
 	}
 }
 
+func (t *Tracker) trackAssignMetadata(ctx context.Context) error {
```

Comment:
We should have some way of turning this off if the mutation controller is turned off, otherwise Gatekeeper will never hit the ready state.

I'm not sure if that's something we need long term or only while mutation is under the "alpha" flag gate

Reply:
Done
Turning it off just like the controller and webhook.